src/main/java/appal/Appal.java

```diff
@@ -0,0 +1,282 @@
+            String line = fileReader.nextLine();
+            String[] words = line.split(COMMA_SEPARATOR);
+            boolean isTaskMarked = words[0].equals(TASK_MARKED_VALUE);
+            String[] taskDetails = Arrays.copyOfRange(words, TASK_INDEX, words.length);
+            handleInput(taskDetails, false);
+            markTask(savedTasksCount, isTaskMarked);
+            savedTasksCount += 1;
+        }
+        Task.setTotalTasks(savedTasksCount);
+    }
+
+    public void loadExistingTasksData() throws NoSavedTasksException{
+        try {
+            loadFileContents();
+        } catch (FileNotFoundException e) {
+            throw new NoSavedTasksException();
+        }
+        printMessage(LOAD_SAVED_TASKS_MESSAGE);
+    }
+
+    public void saveTasksToFile() throws SaveTasksErrorException {
+        try {
+            Files.createDirectories(FILE_DIRECTORY);
+            FileWriter fw = new FileWriter(FILE_PATH); // create a FileWriter in append mode
+            for (int i = 0; i < Task.getTotalTasks(); i++) {
+                fw.write(taskList.get(i).getStatusValue() + COMMA_SEPARATOR + taskList.get(i).getTaskInfo());
+                fw.write(LINE_BREAK);
+            }
+            fw.close();
+        } catch (IOException e) {
+            throw new SaveTasksErrorException();
+        }
+    }
+
+    public void handleInput(String[] inputDetails, boolean isFromUser) {
+        String command = inputDetails[COMMAND_INDEX];
+        try {
+            switch (command) {
+            case COMMAND_BYE:
+                saveTasksToFile();
+                exitAppal();
+                break;
+            case COMMAND_LIST:
+                printTaskList();
+                break;
+            case COMMAND_TODO:
+                addToDo(inputDetails, isFromUser);
+                break;
+            case COMMAND_DEADLINE:
+                addDeadline(inputDetails, isFromUser);
+                break;
+            case COMMAND_EVENT:
+                addEvent(inputDetails, isFromUser);
+                break;
+            case COMMAND_MARK:
+                handleMarkTask(inputDetails, true);
+                break;
+            case COMMAND_UNMARK:
+                handleMarkTask(inputDetails, false);
+                break;
+            case COMMAND_DELETE:
+                deleteTask(inputDetails);
+                break;
+            default:
+                throw new AppalException();
+            }
+        } catch (AppalException e) {
+            printMessage(e.getMessage());
+        }
+    }
+
+    public void runAppal() {
+        welcomeUser();
+        try {
+            loadExistingTasksData();
+        } catch (NoSavedTasksException e) {
+            printMessage(e.getMessage());
+        }
+        while (!isExited) {
+            String line = in.nextLine();
+            String[] inputDetails = Parser.extractInputDetails(line);
+            handleInput(inputDetails, true);
+        }
+    }
+}
```

Comment:
This class seems to be doing a lot; consider extracting related methods into a separate class (e.g., individual command handling)